### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by GhostEsso**
Hello **Code Reviewer :hand:**,
I hope you're doing well :smiley:
In this milestone, I have these features 
- New Classes
- Decorators
- Inheritances
- tested

**Thank you** :pray:  for reviewing my project, **stay blessed** :star2:

### CODE REVIEW

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative 'Nameable'
+
```
Comment (KDlamini) [inline decorator.rb]: Kindly make sure that the `nameable` file is required correctly. In your case, nameable is capitalized which is causing an error in your application.

image

File: capitalize_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require_relative 'decorator'
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name
+  end
+end
```
Comment (KDlamini) [inline capitalize_decorator.rb]: Kindly make sure that the `CapitalizeDecorator` capitalizes the first letter of the given name.

suggestion:

```ruby
...
    def correct_name
      @nameable.correct_name.capitalize
    end
...
```

File: trimmer_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require_relative 'decorator'
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name
+  end
+end
```
Comment (KDlamini) [inline trimmer_decorator.rb]: The trimmer decorate should make sure that the the name has a maximum of 10 characters. If it's longer it should trim the word.

You have submitted a `trimmer_decorator` file that is a duplicate of the `CapitalizeDecorator`. As a result, when testing the features the trimmer decorator doesn't exit.

image

- Kindly make sure that the class is named `TrimmerDecorator` or any relevant class name.
- Please make sure that a condition that trims the name if it is longer than 10 characters is implemented.

suggestion:

```ruby
...
    def correct_name
      @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
    end
...
```

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi @GhostEsso 🙋‍♀️,  

Great  work so far! 💯 
You have done a great job decorating the OOP School Library classes. 🎊 🥳 
There are some issues that you still need to work on to improve your project according to the requirements. You are almost there! 💪🏾 

## Highlights 🌞 

✔️  Good PR title and description
✔️  Professional README file
✔️  Inheritance is well implemented on relevant classes
✔️ Codebase is clean and organized

## Required Changes ♻️
 
**Kindly look through the comments below 👇 for required changes**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- In the future, kindly make sure that the best practices for writing commit messages are followed. One of the best practices is capitalizing the first word of each message like you would any sentence.
- Please make sure that you list meaningful features on your README's key features and future features sections. Make sure that they are relevant to the project.

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: trimmer_decorator.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+require_relative 'decorator'
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
+  end
+end
```
Comment (danmainah) [inline trimmer_decorator.rb]: - [ ] @GhostEsso Great work implementing the changes requested however you need to change the class name from TrimmerDecorator to CapitalizeDecorator, I also noted the output from your tests displays Unknown for the username kindly check your person class and ensure it's working correctly.
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @GhostEsso , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️  Corrected the error when importing the nameable class ,a lso fixed the implementation of the trimmer class.

 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DeliceLydia — APPROVED** [review]
## Hi @GhostEsso, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 

## Highlights
- Addressed changes requested


### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**